Sources/Tracing/Tracer.swift

```diff
@@ -58,72 +59,92 @@ extension Tracer {
 }
 
 // ==== ----------------------------------------------------------------------------------------------------------------
-// MARK: Span creation: with `LoggingContext`
+// MARK: Starting spans: `withSpan`
 
 extension Tracer {
-    /// Start a new `Span` with the given `Baggage` starting at `DispatchWallTime.now()`.
+    /// Execute a specific task within a newly created `Span`.
+    ///
+    /// DO NOT `end()` the passed in span manually. It will be ended automatically when the `operation` returns.
     ///
     /// - Parameters:
     ///   - operationName: The name of the operation being traced. This may be a handler function, database call, ...
-    ///   - context: Logging context containing a `Baggage` which may contain trace identifiers of a parent `Span`.
+    ///   - baggage: Baggage potentially containing trace identifiers of a parent `Span`.
     ///   - kind: The `SpanKind` of the `Span` to be created. Defaults to `.internal`.
-    public func startSpan(
+    ///   - operation: operation to wrap in a span start/end and execute immediately
+    /// - Returns: the value returned by `operation`
+    /// - Throws: the error the `operation` has thrown (if any)
+    public func withSpan<T>(
         _ operationName: String,
-        context: LoggingContext,
-        ofKind kind: SpanKind = .internal
-    ) -> Span {
-        return self.startSpan(operationName, baggage: context.baggage, ofKind: kind, at: .now())
+        baggage: Baggage,
+        ofKind kind: SpanKind = .internal,
+        _ operation: (Span) throws -> T
+    ) rethrows -> T {
+        let span = self.startSpan(operationName, baggage: baggage, ofKind: kind)
```

Comment:
should we add some docs on startSpan that "prefer `withSpan` whenever possible as it automatically manages the spans lifetime; startSpan is provided for APIs which are not able to carry the span from beginning to end enclosed within a closure (e.g. if a span is started in one callback invoked by a 3rd party, and ended in another)."

?